src/light_data.c

```diff
@@ -661,9 +826,38 @@ void update_light_render_area(void)
     light_render_area(startx, starty, endx, endy);
 }
 
-void light_set_light_minimum_size_to_cache(long a1, long a2, long a3)
+void light_set_light_minimum_size_to_cache(long lgt_id, long a2, long a3)
 {
-  _DK_light_set_light_minimum_size_to_cache(a1, a2, a3);
+  // _DK_light_set_light_minimum_size_to_cache(a1, a2, a3);
+  struct Light *lgt;
+  if ( lgt_id )
+  {
+    lgt = &game.lish.lights[lgt_id];
+    if ( lgt->flags & LgtF_Allocated )
+    {
+      if ( lgt->flags & LgtF_Unkn02 )
+      {
+        lgt->flags &= ~2;
+        if ( lgt->flags & LgtF_Dynamic )
+        {
+          lgt->field_9[0] = a2;
+          *(unsigned short *)&lgt->field_1C[8] = a3;
```

Comment:
same.
`byte field_1C[10];` should be split into `byte field_1C[8];` plus `unsigned short field_24;`
and here should be `lgt->field_24 = a3;`